src/main/java/school/hei/patrimoine/visualisation/swing/ihm/google/GoogleDocsSubmitScreen.java

```diff
@@ -156,57 +108,58 @@ private void loadDataInBackground() {
     loadingDialog.setSize(300, 100);
     loadingDialog.setLocationRelativeTo(inputFrame);
 
-    SwingWorker<List<Patrimoine>, Void> worker =
+    SwingWorker<List<NamedString>, Void> worker =
         new SwingWorker<>() {
           @Override
-          protected List<Patrimoine> doInBackground() {
-            var ids = extractInputIds();
-            List<String> codePatrimoinesVisualisables = new ArrayList<>();
-            for (var id : ids) {
-              var code = googleApi.readDocsContent(authDetails, id);
-              codePatrimoinesVisualisables.add(code);
-            }
-            List<Patrimoine> patrimoinesVisualisables = new ArrayList<>();
-            PatrimoineCompiler patrimoineCompiler = new PatrimoineCompiler();
-            for (String codePatrimoine : codePatrimoinesVisualisables) {
-              String className = new ClassNameExtractor().apply(codePatrimoine);
-
-              Patrimoine patrimoineVisualisable =
-                  patrimoineCompiler.apply(className, codePatrimoine);
-
-              patrimoinesVisualisables.add(patrimoineVisualisable);
-            }
-            return patrimoinesVisualisables;
+          protected List<NamedString> doInBackground() {
+            return extractInputData();
           }
 
           @Override
           protected void done() {
             loadingDialog.dispose();
             try {
-              final List<Patrimoine> patrimoinesVisualisables = get();
-              openResultFrame(patrimoinesVisualisables);
+              final List<NamedString> docsLink = get();
```

Comment:
variable name should not be docsLink if it is a list of NamedString

Reply:
done